### PR TITLE
Mapgen: Add watershed mapgen in hidden development form

### PR DESCRIPTION
**Pull request by paramat**
Usable but still in development.

### CODE REVIEW

**Comment by C1ffisme** [comment]
That... is... beautiful...

:+1:

File: src/mapgen_watershed.cpp [hunk 1]
```diff
@@ -0,0 +1,692 @@
...
+	settings->getNoiseParams("mgwatershed_np_mountain",     np_mountain);
+	settings->getNoiseParams("mgwatershed_np_cave1",        np_cave1);
+	settings->getNoiseParams("mgwatershed_np_cave2",        np_cave2);
+	settings->getNoiseParams("mgwatershed_np_cave3",        np_cave3);
+}
+
+
+void MapgenWatershedParams::writeParams(Settings *settings) const
+{
+	settings->setFlagStr("mgwatershed_spflags", spflags, flagdesc_mapgen_watershed, (u32)-1);
```
Comment (kwolekr) [inline src/mapgen_watershed.cpp]: Why don't you use U32_MAX now that we have it?

**Comment by kwolekr** [comment]
Same thing here as with the flat mapgen; you're supposed to update the settingtypes.txt and documentation.

File: src/mapgen_watershed.cpp (hunk 1 above)
Comment (paramat) [inline src/mapgen_watershed.cpp]: Well spotted, thanks.

**Comment by paramat** [comment]
I guess i thought the docs could wait as this is in development, the noise parameters will change many times, however now we have a new settings feature some docs may be necessary, i'll look into that.

**Comment by paramat** [comment]
U32_MAX and U16_MAX now used.

**Comment by 0-afflatus** [comment]
I'd love to see this included when the time is right.

**Pull request closed without merging**